### PR TITLE
MINOR: code cleanup for Kafka Streams task interface

### PR DESCRIPTION
**Pull request by mjsax**
Not functional change. Pure code cleanup.

Call for review @vvcephei (cf. #8852 -- will split up the old PR into multiple smaller ones to simplify reviewing)

### CODE REVIEW

File: streams/src/main/java/org/apache/kafka/streams/processor/internals/AbstractTask.java [hunk 1]
```diff
@@ -124,11 +124,6 @@ public StateStore getStore(final String name) {
         return stateMgr.getStore(name);
     }
 
-    @Override
-    public boolean isClosed() {
```
Comment (mjsax) [inline streams/src/main/java/org/apache/kafka/streams/processor/internals/AbstractTask.java:128]: This method is unused. Removing it.

File: streams/src/main/java/org/apache/kafka/streams/processor/internals/AbstractTask.java [hunk 2]
```diff
@@ -154,7 +149,7 @@ final void transitionTo(final Task.State newState) {
     }
 
     @Override
-    public void update(final Set<TopicPartition> topicPartitions, final Map<String, List<String>> allTopologyNodesToSourceTopics) {
+    public void updateInputPartitions(final Set<TopicPartition> topicPartitions, final Map<String, List<String>> allTopologyNodesToSourceTopics) {
```
Comment (mjsax) [inline streams/src/main/java/org/apache/kafka/streams/processor/internals/AbstractTask.java:152]: Renaming to get better name.

File: streams/src/main/java/org/apache/kafka/streams/processor/internals/Task.java [hunk 3]
```diff
@@ -98,17 +98,9 @@ public boolean isValidTransition(final State newState) {
         }
     }
 
-    TaskId id();
```
Comment (mjsax) [inline streams/src/main/java/org/apache/kafka/streams/processor/internals/Task.java:101]: Group and order methods (no functional change)

File: streams/src/main/java/org/apache/kafka/streams/processor/internals/Task.java [hunk 4]
```diff
@@ -98,17 +98,9 @@ public boolean isValidTransition(final State newState) {
...
-
-    State state();
-
-    default boolean needsInitializationOrRestoration() {
-        return state() == State.CREATED || state() == State.RESTORING;
-    }
 
-    boolean isActive();
 
-    boolean isClosed();
```
Comment (mjsax) [inline streams/src/main/java/org/apache/kafka/streams/processor/internals/Task.java:111]: This method is removed because it's unused.

**Review by vvcephei — APPROVED** [review]
Thanks for this cleanup, @mjsax !

**Pull request merged**